src/App.jsx

```diff
@@ -3,35 +3,48 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
+function App() {
+  //const Is = JSON.parse(localStorage.getItem('todoList')) ?? [];
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
+  const [isLoading, setIsLoading] = useState(true);
 
+  // First effect useEffect to load the initial data
   useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+    const promise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList } });
+      }, 2000);
+    });
+
+    promise.then((result) => {
+      setTodoList(result.data.todoList);
```

Comment:
`setTodoList(result.data.todoList);` works without any issues but it can potentially cause a bug. Since you need to pass a copy of an array to trigger a re-render(think back to useState management) there could be some cases were React does't see this as a mutated array. You already are implementing the solution in anther part of your code...